apps/condo/domains/common/utils/codegeneration/generate.server.utils.js

```diff
@@ -99,11 +104,38 @@ function generateServerUtils (gql) {
         })
     }
 
+    /**
+     * Returns result with flat related IDs, relations would not be resolved because of getById
+     * @param context
+     * @param attrs
+     * @param existingItem
+     * @returns {Promise<*|null|undefined>}
+     */
+    async function createOrUpdateExistingItem (context, attrs, existingItem) {
+        let id = get(existingItem, 'id', null)
+
+        if (existingItem) {
+            await update(context, id, attrs)
+        } else {
+            const createdItem = await create(context, attrs)
+
+            id = createdItem.id
+        }
+
+        // NOTE: result will have flat related IDs, relations not resolved because of getById
+        return await getById(gql.SINGULAR_FORM, id)
+    }
+
+
```

Comment:
This will bloat core API with logically identical operations and it will be harder to work with.
As a solution, I would try to add fields customisation option to functions, returned by `generateServerUtils`.

Reply:
fixed